pkg/collector/upgrade/v0_104_0.go

```diff
@@ -37,3 +38,34 @@ func upgrade0_104_0(u VersionUpgrade, otelcol *v1beta1.OpenTelemetryCollector) (
 	u.Recorder.Event(otelcol, "Warning", "Upgrade", warnStr)
 	return otelcol, nil
 }
+
+// TAUnifyEnvVarExpansion disables confmap.unifyEnvVarExpansion featuregate on
+// collector instances if a prometheus receiver is configured.
+// NOTE: We need this for now until 0.105.0 is out with this fix:
+// https://github.com/open-telemetry/opentelemetry-collector/commit/637b1f42fcb7cbb7ef8a50dcf41d0a089623a8b7
+func TAUnifyEnvVarExpansion(otelcol *v1beta1.OpenTelemetryCollector) {
```

Comment:
Should we add something in the upgrade routine to 0.105.0 to remove the flag if it exists?

Reply:
+1